File: frontend/src/types/inbox.ts
export interface InboxAttachment {
  filename: string;
  content_type: string;
  size: number;
  content_base64?: string;
}

export interface InboxEmailSummary {
  id: string;
  from_address: string;
  to_addresses: string[];
  subject: string | null;
  is_read: boolean;
  received_at: string;
  raw_size: number;
  has_attachments: boolean;
}

export interface InboxEmail {
  id: string;
  workspace_id: string;
  message_id: string | null;
  from_address: string;
  to_addresses: string[];
  cc_addresses: string[];
  bcc_addresses: string[];
  subject: string | null;
  text_body: string | null;
  html_body: string | null;
  headers: Record<string, string>;
  attachments: InboxAttachment[];
  raw_size: number;
  is_read: boolean;
  received_at: string;
}

export interface InboxEmailSummaryList {
  data: InboxEmailSummary[];
  total: number;
  unread_count: number;
}

export interface SmtpCredentials {
  smtp_host: string;
  smtp_port: number;
  smtp_username: string;
  smtp_password: string;
  connection_url: string;
}

File: backend/app/schemas/inbox.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class InboxAttachment(BaseModel):
    filename: str
    content_type: str
    size: int
    content_base64: str | None = None  # Only included in detail view


class InboxEmailSummary(BaseModel):
    id: uuid.UUID
    from_address: str
    to_addresses: list[str]
    subject: str | None
    is_read: bool
    received_at: datetime
    raw_size: int
    has_attachments: bool

    model_config = {"from_attributes": True}


class InboxEmailResponse(BaseModel):
    id: uuid.UUID
    workspace_id: uuid.UUID
    message_id: str | None
    from_address: str
    to_addresses: list[str]
    cc_addresses: list[str]
    bcc_addresses: list[str]
    subject: str | None
    text_body: str | None
    html_body: str | None
    headers: dict
    attachments: list[InboxAttachment]
    raw_size: int
    is_read: bool
    received_at: datetime

    model_config = {"from_attributes": True}


class InboxEmailSummaryListResponse(BaseModel):
    data: list[InboxEmailSummary]
    total: int
    unread_count: int


class SmtpCredentialsResponse(BaseModel):
    smtp_host: str
    smtp_port: int
    smtp_username: str
    smtp_password: str
    connection_url: str
